Tools/TestWebKitAPI/Tests/WebCore/PublicSuffix.cpp

```diff
@@ -75,112 +76,113 @@ TEST_F(PublicSuffix, IsPublicSuffix)
+    EXPECT_EQ(String("com.com"_s), publicSuffixStore.topPrivatelyControlledDomain("www.com.com"_s));
+    EXPECT_EQ(String("test.co.uk"_s), publicSuffixStore.topPrivatelyControlledDomain("test.co.uk"_s));
+    EXPECT_EQ(String("test.co.uk"_s), publicSuffixStore.topPrivatelyControlledDomain("subdomain.test.co.uk"_s));
+    EXPECT_EQ(String("bl.uk"_s), publicSuffixStore.topPrivatelyControlledDomain("bl.uk"_s));
+    EXPECT_EQ(String("bl.uk"_s), publicSuffixStore.topPrivatelyControlledDomain("subdomain.bl.uk"_s));
+    EXPECT_EQ(String("test.xn--zf0ao64a.tw"_s), publicSuffixStore.topPrivatelyControlledDomain("test.xn--zf0ao64a.tw"_s));
+    EXPECT_EQ(String("test.xn--zf0ao64a.tw"_s), publicSuffixStore.topPrivatelyControlledDomain("www.test.xn--zf0ao64a.tw"_s));
+    EXPECT_EQ(String("127.0.0.1"_s), publicSuffixStore.topPrivatelyControlledDomain("127.0.0.1"_s));
+    EXPECT_EQ(String(), publicSuffixStore.topPrivatelyControlledDomain("1"_s));
+    EXPECT_EQ(String(), publicSuffixStore.topPrivatelyControlledDomain("com"_s));
+    EXPECT_EQ(String("test.com"_s), publicSuffixStore.topPrivatelyControlledDomain("r4---asdf.test.com"_s));
+    EXPECT_EQ(String("r4---asdf.com"_s), publicSuffixStore.topPrivatelyControlledDomain("r4---asdf.com"_s));
+    EXPECT_EQ(String(), publicSuffixStore.topPrivatelyControlledDomain("r4---asdf"_s));
     EXPECT_EQ(utf16String(bidirectionalDomain), utf16String(bidirectionalDomain));
-    EXPECT_EQ(String("example.com"_s), topPrivatelyControlledDomain("ExamPle.com"_s));
-    EXPECT_EQ(String("example.com"_s), topPrivatelyControlledDomain("SUB.dOmain.ExamPle.com"_s));
-    EXPECT_EQ(String("localhost"_s), topPrivatelyControlledDomain("localhost"_s));
-    EXPECT_EQ(String("localhost"_s), topPrivatelyControlledDomain("LocalHost"_s));
-    EXPECT_EQ(String::fromUTF8("åäö"), topPrivatelyControlledDomain(String::fromUTF8("åäö")));
-    EXPECT_EQ(String::fromUTF8("ÅÄÖ"), topPrivatelyControlledDomain(String::fromUTF8("ÅÄÖ")));
-    EXPECT_EQ(String("test.com"_s), topPrivatelyControlledDomain(".test.com"_s));
-    EXPECT_EQ(String(), topPrivatelyControlledDomain("...."_s));
+    EXPECT_EQ(String("example.com"_s), publicSuffixStore.topPrivatelyControlledDomain("ExamPle.com"_s));
+    EXPECT_EQ(String("example.com"_s), publicSuffixStore.topPrivatelyControlledDomain("SUB.dOmain.ExamPle.com"_s));
+    EXPECT_EQ(String("localhost"_s), publicSuffixStore.topPrivatelyControlledDomain("localhost"_s));
+    EXPECT_EQ(String("localhost"_s), publicSuffixStore.topPrivatelyControlledDomain("LocalHost"_s));
+    EXPECT_EQ(String::fromUTF8("åäö"), publicSuffixStore.topPrivatelyControlledDomain(String::fromUTF8("åäö")));
+    EXPECT_EQ(String::fromUTF8("ÅÄÖ"), publicSuffixStore.topPrivatelyControlledDomain(String::fromUTF8("ÅÄÖ")));
+    EXPECT_EQ(String("test.com"_s), publicSuffixStore.topPrivatelyControlledDomain(".test.com"_s));
+    EXPECT_EQ(String(), publicSuffixStore.topPrivatelyControlledDomain("...."_s));
 }
```

Comment:
You should add tests that utilize the new `enablePublicSuffixCache` and `addPublicSuffix` functionality.